ec_deployment: add support for data tiers (#253)
elastic/terraform-provider-ec: docs/resources/ec_deployment.md

The optional `elasticsearch.topology` block supports the following arguments:

> * `id` - (Required) Unique topology identifier, this will generally refer to an Elasticsearch data tier, such as `hot_content`, `warm` or `cold`.

Reviewer: ```suggestion
* `id` - (Required) Unique topology identifier. It generally refers to an Elasticsearch data tier, such as `hot_content`, `warm` or `cold`.
```

* `size` - (Optional) Amount in Gigabytes per topology element in the `"<size in GB>g"` notation. When omitted, it defaults to the deployment template value.
* `size_resource` - (Optional) Type of resource to which the size is assigned. Defaults to `"memory"`.
* `zone_count` - (Optional) Number of zones the instance type of the Elasticsearch cluster will span. This is used to set or unset HA on an Elasticsearch node type. When omitted, it defaults to the deployment template value.
* `node_type_data` - (Optional) The node type for the Elasticsearch cluster (data node).
* `node_type_master` - (Optional) The node type for the Elasticsearch cluster (master node).
* `node_type_ingest` - (Optional) The node type for the Elasticsearch cluster (ingest node).
* `node_type_ml` - (Optional) The node type for the Elasticsearch cluster (machine learning node).

> ~> **Note when node_type_* fields set** After upgrading to a version that supports data tiers (7.10.0 or above), the `node_type_*` will not have any effect even if specified. The provider will automatically migrate the `node_type_*` fields to the appropriate `node_roles` as set by the deployment template. After having upgraded to `7.10.0` or above, the fields should be removed from the terraform configuration if explicitly configured.

Reviewer: ```suggestion
~> **Note when node_type_* fields set** After upgrading to a version that supports data tiers (7.10.0 or above), the `node_type_*` has no effect even if specified. The provider automatically migrates the `node_type_*` fields to the appropriate `node_roles` as set by the deployment template. After having upgraded to `7.10.0` or above, the fields should be removed from the terraform configuration, if explicitly configured.
```

* `elasticsearch.#.cloud_id` - Encoded Elasticsearch credentials to use in Beats or Logstash. For more information, see [Configure Beats and Logstash with Cloud ID](https://www.elastic.co/guide/en/cloud/current/ec-cloud-id.html).
* `elasticsearch.#.http_endpoint` - Elasticsearch resource HTTP endpoint.
* `elasticsearch.#.https_endpoint` - Elasticsearch resource HTTPs endpoint.
* `elasticsearch.#.topology.#.instance_configuration_id` - instance configuration of the deployment topology element.
* `elasticsearch.#.topology.#.node_type_data` - Node type (data) for the Elasticsearch topology element.
* `elasticsearch.#.topology.#.node_type_master` - Node type (master) for the Elasticsearch topology element.
* `elasticsearch.#.topology.#.node_type_ingest` - Node type (ingest) for the Elasticsearch topology element.
* `elasticsearch.#.topology.#.node_type_ml` - Node type (machine learning) for the Elasticsearch topology element.

> * `elasticsearch.#.topology.#.node_roles` - List of roles for the the topology element. They are inferred from the deployment template.

Reviewer: ```suggestion
* `elasticsearch.#.topology.#.node_roles` - List of roles for the topology element. They are inferred from the deployment template.
```
